extensions/omni_httpd/master_worker.c

```diff
@@ -511,15 +511,34 @@ void master_worker(Datum db_oid) {
       strncpy(worker.bgw_library_name, MyBgworkerEntry->bgw_library_name, BGW_MAXLEN - 1);
       for (int i = 0; i < *num_http_workers; i++) {
         BackgroundWorkerHandle *handle;
-        RegisterDynamicBackgroundWorker(&worker, &handle);
-        pid_t worker_pid;
-        if (WaitForBackgroundWorkerStartup(handle, &worker_pid) == BGWH_POSTMASTER_DIED) {
-          goto terminate;
+        if (RegisterDynamicBackgroundWorker(&worker, &handle)) {
+          pid_t worker_pid;
+          BgwHandleStatus status = WaitForBackgroundWorkerStartup(handle, &worker_pid);
+
+          switch (status) {
+          case BGWH_POSTMASTER_DIED:
+            goto terminate;
+          case BGWH_STARTED: {
+            cvec_bgwhandle_push(&http_workers, handle);
+            break;
+          }
+          default:
+            ereport(ERROR, (errcode(ERRCODE_INSUFFICIENT_RESOURCES),
+                            errmsg("Could not start %s", worker.bgw_name)));
```

Comment:
I've also added this case considering that auto_prewarm does the same:
https://github.com/postgres/postgres/blob/dc0178a7fae05915ca3ae3f60ab89dad15dd6d2e/contrib/pg_prewarm/autoprewarm.c#L830-L841